orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy

```diff
@@ -61,30 +63,80 @@ trait DeploymentDetailsAware {
     }
   }
 
-  List<Stage> getAncestors(Stage stage) {
-    if (stage.requisiteStageRefIds) {
-      def previousStages = stage.execution.stages.findAll {
+  List getPipelineExecutions(def execution) {
+    // The initial stage execution is a Pipeline, and the ancestor executions are Maps.
+    if (execution in Pipeline || execution?.context) {
+      return [execution] + getPipelineExecutions(getParentPipelineExecution(execution))
+    } else {
+      return []
+    }
+  }
+
+  List<Stage> getAncestors(def stage, def execution) {
+    if (stage?.requisiteStageRefIds) {
+      def previousStages = execution.stages.findAll {
         it.refId in stage.requisiteStageRefIds
       }
-      def syntheticStages = stage.execution.stages.findAll {
+      def syntheticStages = execution.stages.findAll {
         it.parentStageId in previousStages*.id
       }
-      return (previousStages + syntheticStages) + previousStages.collect { getAncestors(it) }.flatten()
-    } else if (stage.parentStageId) {
-      def parent = stage.execution.stages.find { it.id == stage.parentStageId }
-      return ([parent] + getAncestors(parent)).flatten()
+      return (previousStages + syntheticStages) + previousStages.collect { getAncestors(it, execution) }.flatten()
+    } else if (stage?.parentStageId) {
+      def parent = execution.stages.find { it.id == stage.parentStageId }
+      return ([parent] + getAncestors(parent, execution)).flatten()
     } else {
+      def parentPipelineExecution = getParentPipelineExecution(execution)
+
+      if (parentPipelineExecution) {
+        def parentPipelineStageId = execution.trigger?.parentPipelineStageId
+
+        def parentPipelineStage = parentPipelineExecution.stages?.find {
+          it.type == "pipeline" && it.id == parentPipelineStageId
+        }
+
+        if (parentPipelineStage) {
+          return getAncestors(parentPipelineStage, parentPipelineExecution)
+        } else {
+          def parentPipelineStages = parentPipelineExecution.stages?.collect()?.sort {
+            a, b -> b.endTime <=> a.endTime
+          }
+
+          if (parentPipelineStages) {
+            // The list is sorted in reverse order by endTime.
+            def firstStage = parentPipelineStages.last()
+
+            return parentPipelineStages + getAncestors(firstStage, parentPipelineExecution)
+          } else {
+            // Parent pipeline has no stages.
+            return getAncestors(null, parentPipelineExecution)
+          }
+        }
+      }
+
       return []
     }
   }
 
+  private getParentPipelineExecution(def execution) {
```

Comment:
Sorry, I missed the bit about dealing with a `Map` as well. 

That's coming from `execution.trigger.parentExecution`? 

In my mind I was thinking we would fetch the parent pipeline out of the `ExecutionRepository` on-demand rather than always expecting it to be passed around as `parentExecution` ... might just be me tho. 

If we can't do ^^, no change to `ObjectMapper.convert` the parentExecution to a proper Pipeline. I can almost guarantee we're going to get hit with an issue related to the extremely loose typing in here.